File: backend/Models/Comment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Backend.Models{
    public class Comment{
        [Key]
        public int commentId {get; set;}
        [Required(ErrorMessage = "Description is required.")]
        [StringLength(maximumLength: 3000, MinimumLength = 1, ErrorMessage = "Description must be between 1 and 3000 characters.")]
        public required string description {get; set;}
        [Required(ErrorMessage = "UserId is required.")]
        public bool isDeleted {get; set; }
        [Required(ErrorMessage = "UserId is required.")]
        public int userId {get; set;}
        [Required(ErrorMessage = "PostId is required.")]
        public int postId {get; set;}
        public DateTime? createdDate {get; set;}
        public DateTime? modifiedDate {get; set;}
        public DateTime? deletedDate { get; set; }
        [ForeignKey("userId")]
        [JsonIgnore]
        public User? commenter {get; set;}
        [ForeignKey("postId")]
        [JsonIgnore]
        public required Post post {get; set;}
    }
}

File: frontend/src/app/models/comments/comment.ts
export class Comment {
  #id!: string;
  #userId!: string;
  #username!: string;
  #userProfile!: string;
  #text!: string;
  #createdDate!: Date;

  get id(): string {
    return this.#id;
  }

  get userId(): string {
    return this.#userId;
  }

  get username(): string {
    return this.#username;
  }

  get userProfile(): string {
    return this.#userProfile;
  }

  get text(): string {
    return this.#text;
  }

  get createdDate(): Date {
    return this.#createdDate;
  }

  // Setters
  set id(value: string) {
    this.#id = value;
  }

  set userId(value: string) {
    this.#userId = value;
  }

  set username(value: string) {
    this.#username = value;
  }

  set userProfile(value: string) {
    this.#userProfile = value;
  }

  set text(value: string) {
    this.#text = value;
  }

  set createdDate(value: Date) {
    this.#createdDate = value;
  }

  static fromJSON(json: any): Comment {
    const comment = new Comment();
    comment.id = json.id;
    comment.userId = json.userId;
    comment.username = json.username;
    comment.userProfile = json.userProfile;
    comment.text = json.text;
    comment.createdDate = new Date(json.createdDate);
    return comment;
  }
}
